packages/components/src/components/FileUploader/FileUploader.tsx

```diff
@@ -0,0 +1,282 @@
+import { Box } from "../../primitives/Box";
+import { Text } from "../../primitives/Text";
+import { Anchor } from "../Anchor";
+import { Button, ButtonProps } from "../Button";
+import { Icon } from "../Icon";
+import { HelpText } from "../HelpText";
+import { ProgressBar } from "../ProgressBar";
+import { FileUploaderButtonProps } from "./FileUploaderButton";
+
+export interface FileUploaderProps
+  extends React.HTMLAttributes<HTMLDivElement> {
+  /** The title of the file */
+  label: string;
+
+  /** The call to action button */
+  children: ReactElement<ButtonProps | FileUploaderButtonProps>;
+
+  /** The max size of a file */
+  maxFileSize?: string;
+
+  /** The file name */
+  fileName?: string;
+
+  /** The file url when the file is uploaded */
+  fileUrl?: string;
+
+  /** The size of the file */
+  fileSize?: string;
+
+  /** An error message */
+  errorMessage?: string;
+
+  /** The upload progress */
+  progress?: number;
+
+  /** Handles cancel upload  */
+  onCancel?: () => void;
+
+  /** Handles remove file */
+  onRemove?: () => void;
+}
+
+type FileUploaderStatus = "error" | "loading" | "success" | "waiting";
+
+const getIconContainerProps = (
+  status: FileUploaderStatus
+): {
+  backgroundColor: SystemProp<keyof Theme["colors"], Theme>;
+} => {
+  switch (status) {
+    case "success": {
+      return { backgroundColor: "colorBackgroundSuccess" };
+    }
+    case "error": {
+      return { backgroundColor: "colorBackgroundError" };
+    }
+    default: {
+      return { backgroundColor: "colorBackgroundInfo" };
+    }
+  }
+};
+
+const getIconProps = (
+  status: FileUploaderStatus
+): {
+  color: SystemProp<keyof Theme["colors"], Theme>;
+  icon: keyof typeof HeroOutlineIcons;
+} => {
+  switch (status) {
+    case "success": {
+      return { icon: "PaperClipIcon", color: "colorTextSuccess" };
+    }
+    case "error": {
+      return { icon: "XCircleIcon", color: "colorTextError" };
+    }
+    default: {
+      return { icon: "CloudArrowUpIcon", color: "colorTextLink" };
+    }
+  }
+};
+
+const getStatus = (
+  progress: number,
+  fileUrl?: string,
+  errorMessage?: string
+): FileUploaderStatus => {
+  if (progress > 0 && progress < 100) return "loading";
+  if (!isEmpty(fileUrl)) return "success";
+  if (!isEmpty(errorMessage)) return "error";
+  return "waiting";
+};
+
+const CancelUploadButton = (props: Partial<ButtonProps>): ReactElement => {
+  const isMobile = useDown("md");
+
+  if (isMobile) {
+    return (
+      <Button {...props} variant="secondary">
+        Cancel upload
+      </Button>
+    );
+  }
+
+  return (
+    <Button
+      {...props}
+      aria-label="Cancel upload"
+      iconOnly
+      trailingIcon="XMarkIcon"
+      variant="ghost"
+    />
+  );
+};
+
+const RemoveButton = (props: Partial<ButtonProps>): ReactElement => {
+  const isMobile = useDown("md");
+
+  if (isMobile) {
+    return (
+      <Button {...props} aria-label="Remove file" variant="secondary">
```

Comment:
We need to add the `onRemove` prop here.

Reply:
Is this necessary here? We already have the onRemove on the parent component https://github.com/Localitos/pluto/blob/4afcf99ab45deb2bed2feddd09b01e137c45ea1b/packages/components/src/components/FileUploader/FileUploader.tsx#L265-L267